File: src/main/java/ua/softgroup/matrix/server/persistent/repository/ProjectRepository.java
package ua.softgroup.matrix.server.persistent.repository;

import org.springframework.data.repository.CrudRepository;
import ua.softgroup.matrix.server.persistent.entity.Project;
import ua.softgroup.matrix.server.persistent.entity.User;

import java.util.List;
import java.util.Set;

public interface ProjectRepository extends CrudRepository<Project, Long> {

    Project findBySupervisorIdAndUser(Long supervisorId, User user);
    
    List<Project> findByUser(User user);

    Set<Project> findByUserId(Long userId);

    Set<Project> findBySupervisorId(Long supervisorId);
}

File: src/main/scala/ua/softgroup/matrix/server/persistent/repository/ProjectRepository.scala
package ua.softgroup.matrix.server.persistent.repository

import java.util

import org.springframework.data.repository.CrudRepository
import ua.softgroup.matrix.server.persistent.entity.{Project, User}

/**
  * @author Oleksandr Tyshkovets <sg.olexander@gmail.com> 
  */
trait ProjectRepository extends CrudRepository[Project, java.lang.Long] {

  def findBySupervisorIdAndUser(supervisorId: Long, user: User): Project

  def findByUser(user: User): util.List[Project]

  def findByUserId(userId: Long): util.Set[Project]

  def findBySupervisorId(supervisorId: Long): util.Set[Project]

}
